iac-extensions/arm/src/test/resources/checks/IpRestrictedAdminAccessCheck/Microsoft.Network_networkSecurityGroups_securityRules/protocol.bicep

```diff
@@ -0,0 +1,57 @@
+// Noncompliant@+1
+resource raise_an_issue_sensitive_case_protocol_values_1 'Microsoft.Network/networkSecurityGroups/securityRules@2022-11-01' = {
+  name: 'Raise an issue: sensitive case protocol values 1'
+  properties: {
+    direction: 'Inbound'
+    access: 'Allow'
+    protocol: 'TCP'
+    destinationPortRange: '22'
+    sourceAddressPrefix: 'Internet'
+  }
+}
+
+// Noncompliant@+1
+resource raise_an_issue_sensitive_case_protocol_values_2 'Microsoft.Network/networkSecurityGroups/securityRules@2022-11-01' = {
+  name: 'Raise an issue: sensitive case protocol values 2'
+  properties: {
+    direction: 'Inbound'
+    access: 'Allow'
+    protocol: 'tcp'
+    destinationPortRange: '22'
+    sourceAddressPrefix: 'Internet'
+  }
+}
+
+// Noncompliant@+1
+resource raise_an_issue_sensitive_case_protocol_values_3 'Microsoft.Network/networkSecurityGroups/securityRules@2022-11-01' = {
+  name: 'Raise an issue: sensitive case protocol values 3'
+  properties: {
+    direction: 'Inbound'
+    access: 'Allow'
+    protocol: '*'
+    destinationPortRange: '22'
+    sourceAddressPrefix: 'Internet'
+  }
+}
+
+resource compliant_protocol_values_1 'Microsoft.Network/networkSecurityGroups/securityRules@2022-11-01' = {
+  name: 'Compliant: protocol values 1'
+  properties: {
+    direction: 'Inbound'
+    access: 'Allow'
+    protocol: 'UDP'
+    destinationPortRange: '22'
+    sourceAddressPrefix: 'Internet'
+  }
+}
+
+resource compliant_protocol_values_2 'Microsoft.Network/networkSecurityGroups/securityRules@2022-11-01' = {
+  name: 'Compliant: protocol values 2'
+  properties: {
+    direction: 'Inbound'
+    access: 'Allow'
+    protocol: ''
+    destinationPortRange: '22'
+    sourceAddressPrefix: 'Internet'
+  }
+}
```

Comment:
Please add similar test case for `Microsoft.Network_networkSecurityGroups_securityRules/other.json`

Reply:
I've seen that test, but I'm not sure how relevant it is, since it actually checks the logic of `AbstractArmResourceCheck` - that check isn't called for a resource of the wrong type. Or am I missing something?

Reply:
With such tests, we verify that logic takes into account the type of resource.
This prevents us from introducing some regression.